static/src/javascripts/projects/common/modules/commercial/creatives/hosted-thrasher.js

```diff
@@ -0,0 +1,41 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/template',
+    'common/modules/commercial/hosted-video',
+    'common/modules/commercial/creatives/add-tracking-pixel',
+    'text!common/views/commercial/creatives/hosted-thrasher.html'
+], function (
+    fastdom,
+    template,
+    hostedVideo,
+    addTrackingPixel,
+    hostedVideoStr
+) {
+    var hostedVideoTpl;
+
+    var HostedThrasher = function ($adSlot, params) {
+        this.$adSlot = $adSlot;
+        this.params = params;
+    };
+
+    HostedThrasher.prototype.create = function () {
+        if (!hostedVideoTpl) {
+            hostedVideoTpl = template(hostedVideoStr);
+        }
+
+        var domPromise = fastdom.write(function () {
+            this.$adSlot.append(hostedVideoTpl({ data: this.params}));
+        }.bind(this));
```

Comment:
Luckily, the objects created from these prototypes get GC'd pretty quickly (they're never assigned); thus, even though this is terrible API design, there is no memory leak. Oliver's design keeps internal state, so there is room for improvement.